ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -111,15 +135,53 @@ message AllocatePartyRequest {
 
   // Human-readable name of the party to be added to the participant. It doesn't
   // have to be unique.
+  // Use of this field is discouraged. Consider using modifiable participant local party metadata instead.
   // Optional
   string display_name = 2;
+
+  // Participant local metadata to be bound to this newly allocated party.
+  // Optional
+  ObjectMeta metadata = 3;
 }
 
 message AllocatePartyResponse {
 
   PartyDetails party_details = 1;
 }
 
+// The participant local metadata of a party.
+// This metadata is stored locally on the backing participant and is not exposed on the given ledger.
+message PartyMetadata {
+
+  // The stable unique identifier of a Daml party.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string party = 1;
+
+  // The metadata of this party.
+  // Optional
+  ObjectMeta metadata = 2;
+}
```

Comment:
Why not extend `PartyDetails` instead?

Reply:
It's an attempt to separate a party on a ledger (`PartyDetails`) from a participant local party (`PartyMetadata`).
`PartyMetadata` is what we have closest to the `User` resource: it's participant local and in principle it's modifiable.
 
Also the update rpc
```
rpc UpdatePartyMetadata (UpdatePartyMetadataRequest) returns (UpdatePartyMetadataResponse);
```
is about updating participant local metadata of a party, so it seems clean that it's separated from the on-ledger party details.


We still include `ObjectMeta metadata` in `PartyDetails` message but that's rather for convenience

Reply:
For the record, we dropped `PartyMetadata` and are using `PartyDetails`